rpmbuild/bin/copr-builder-ready

```diff
@@ -0,0 +1,100 @@
+#! /usr/bin/python3
+
+"""
+Final checks that the builder machine is ready to be used
+
+Everything printed to STDOUT will be redirected to the copr-backend logs,
+STDERR will be ignored.
+"""
+
+import os
+import sys
+import time
+from fnmatch import fnmatch
+from subprocess import run, PIPE
+from copr_rpmbuild.config import Config
+
+
+def check_mock_config(chroot):
+    """
+    Does the mock config for this chroot exist?
+    """
+    if chroot == "srpm-builds":
+        return
+
+    config = "/etc/mock/{}.cfg".format(chroot)
+    if os.path.isfile(config):
+        return
+
+    print("Chroot config {} not found".format(config))
+    sys.exit(1)
+
+
+def subscription_required(chroot):
+    """
+    Is subscription required for this task?
+    """
+    config = Config()
+    config.load_config()
+
+    for pattern in config.rhsm:
+        if fnmatch(chroot, pattern):
+            return True
+    return False
+
+
+def active_subscription():
+    """
+    Is subscription active on this system?
+    """
+    # This implementation requires root privileges which we fortunately have
+    # when calling this script. In case this function needs to be re-written to
+    # be used under a normal user, it could be done by checking the existence of
+    # the `/etc/pki/consumer/cert.pem` file. However, it will break in the
+    # following corner cases:
+    #   - The system is registered and then halted for a long time and then
+    #     booted again after the entitlement is no longer valid
+    #   - The system is unregistered from the server (but not from the client
+    #     itself), making all the client files stale
+    cmd = ["subscription-manager", "status"]
+    return run(cmd, stdout=PIPE, stderr=PIPE, check=False) == 0
```

Comment:
There are two operations ... register and attach. Does this confirm that we are attached and can consume the repos?